doc/source/dev/contributor/quickstart_ubuntu.rst

```diff
@@ -70,12 +72,6 @@ Building SciPy
 
 #. (Optional) Check your present working directory by entering ``pwd`` at the terminal. You should be in the root ``..../scipy`` directory, not in a directory ending ``..../scipy/scipy``.
 
-#. Install `Homebrew on Linux`_. Enter into the terminal |br| ``sh -c "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install.sh)"`` |br| or follow the installation instructions listed on the `Homebrew on Linux`_ website. Homebrew requires additional packages to be installed, and lists the required commands in the terminal window. Copy and paste these commands into the terminal to complete the Homebrew setup. If no additional commands are given, they may be found `here <https://docs.brew.sh/Homebrew-on-Linux>`_.
-
-   Homebrew is a package manager that will help you download ``gcc``, the software we will use to compile C, C++, and Fortran code included in SciPy.
-
-#. Use Homebrew to install ``gcc`` by entering the command ``brew install gcc``.
-
 #. Do an in-place build: enter ``python3 setup.py build_ext --inplace``. |br| This will compile the C, C++, and Fortran code that comes with SciPy. We installed ``python3`` with Anaconda. ``setup.py`` is a script in the root directory of SciPy, which is why you have to be in the SciPy root directory to call it. ``build_ext`` is a command defined in ``setup.py``, and ``--inplace`` is an option we'll use to ensure that the compiling happens in the SciPy directory you already have rather than the default location for Python packages. By building in-place, you avoid having to re-build SciPy before you can test changes to the Python code.
 
```

Comment:
Why an inplace build here?  I think `python setup.py build && python runtests.py` builds scipy only once. 

Anyway, it's a small matter, feel free to ignore

Reply:
If you don't do an in-place build, don't you have to build every time you want to test changes?

Reply:
Not with `runtests.py`, I think. Subsequent rebuilds are much faster, so I suspect the sys.path manipulations in runtests are equivalent to an inplace build.

Reply:
What if you are running a script other than `runtests.py` or testing changes interactively in an IDE?

Reply:
> or testing changes interactively in an IDE? 

I think that's what `runtests.py --ipython` is for. In an IDE you'd have to set PYTHONPATH manually.

I think `runtests.py` has pros and cons; I've never been much of a fan, but it does make life easier for beginning contributors for most development tasks.

Reply:
I thought the point of `--inplace` is you can import extension modules directly in the project directory without installing the whole library. However, SciPy expressly forbids doing that:

https://github.com/scipy/scipy/blob/28d4244b6f5c1543755e586f27f9261d0f84153b/scipy/__init__.py#L122-L124

If that's right, then I don't think you'd ever want to build SciPy inplace.

If you're working on native code `runtests.py` is excellent because you can't forget to rebuild before testing. If you only work on python then I guess `python setup.py develop` would be just as good?

Reply:
OK. I can't make that change, though, because I don't know about this feature of `runtests.py`.

Reply:
TBH `runtests.py --ipython` is so huge relief compared to fiddling with PYTHONPATH or `setup.py develop` or forgetting to `setup.py install` etc

workflow wise I typically do inplace builds for small projects. When dealing with beginners, `setup.py build --i place` is generally a great suggestion --- with an explicit exception of building numpy/scipy where runtests.py is already set up.

So I'd suggest recommending runtests in the beginner quickstart guide for building scipy specifically, and mention inplace in a more general "working with compiled extensions" part of the guide.

Reply:
> In an IDE you'd have to set PYTHONPATH manually.

`conda develop .`

Reply:
> I thought the point of `--inplace` is you can import extension modules directly in the project directory without installing the whole library. However, SciPy expressly forbids doing that:

No that's not it. My typical workflow is `python setup.py build_ext -i` followed by (in another shell from anywhere, or from an editor/IDE, with `PYTHONPATH` pointing to repo root) importing and testing. It's faster iterating than with `runtests.py`, and one can use any tool one wants rather than being limited to what `runtests.py` supports as a command-line flag.

Reply:
@rgommers what benefit do you get specifically from the `-i` flag though?

Reply:
1. That I know where my build is (typically `~/code/scipy`), rather than in some new directory that `runtests.py` created
2. If I want to edit Python code in SciPy, I can just have edit-test cycles (e.g. in a Jupyter notebook), rather than "edit, run-runtests-to-copy-edited-code-over, test" cycles.
3. I want to see my build log as it goes

To me, `runtests.py` is nice for beginners because it abstracts some details. Once you know those details though, the abstraction layer just gets in the way.

Reply:
1: It's not `runtests.py` that creates the folder, it's `distutils`. Out of source builds are the default because it's easier to clean up build artefacts and you can build multiple incompatible configurations without interfering with the other's build artefacts. e.g. building for multiple python versions in the same directory without cleaning in between.

2 & 3: The same is true for `setup.py develop`, no?

Reply:
1. Well, that's semantics - `runtests.py` has extensive `PYTHONPATH` manipulation and building with `'--prefix=' , '--single-version-externally-managed',` to control where the build ends up.

> 2 & 3: The same is true for setup.py develop, no?

Yes exactly. `build_ext -i` and `develop` are effectively the same, the only difference is that `develop` makes an entry in `site-packages` so you should use that if you have a separate development environment only, while `build_ext -i` you can use even in your main env where you have Jupyter, Dask and whatever else you want to play with installed. I tend to do the latter more than the former, since remembering to do `conda activate scipy-dev` each time I open a new terminal is a pain.

Reply:
Ah so maybe recommend `develop` to people using environments and `build_ext -i` if using the system python?

Reply:
As long as it's a quickstart tutorial, I guess it's best to pick _a_ recommendation and stick to it. And link to a separate "possible workflows" document which lists various workflows.

Reply:
The intent of this PR was to make simplifications based on the conda compilers working. This PR didn't add the `--inplace` option, so unless there is consensus, maybe we leave this to a separate issue/PR?